drivers/CoreIMU/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Leka - LekaOS
+# Copyright 2022 APF France handicap
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(CoreIMU STATIC)
+
+target_include_directories(CoreIMU
+	PUBLIC
+	include
+	extern/include
+)
+
+target_sources(CoreIMU
+	PRIVATE
+	extern/source/lsm6dsox_reg.c
+	source/CoreLSM6DSOX.cpp
+	source/CoreAccelerometer.cpp
+	source/CoreGyroscope.cpp
+)
+
+target_link_libraries(CoreIMU)
+
+if(${CMAKE_PROJECT_NAME} STREQUAL "LekaOSUnitTests")
+	leka_unit_tests_sources(
```

Comment:
Some advice about unit tests

- `CoreHTS` can be an inspiration since it follows the same pattern (I2C, ST driver, io_writer defines out of driver, etc.)  
- The only thing you can spy/mock is I2C (that's why it's important to use `CoreI2C` and not the mbed one).
  - Since there, for the most of methods you just have to check that `read` and `write` are in a correct quantity.
  - However, some test require attention, like `setPowerMode` (does the `write` has the correct buffer?) and `updateData` (return a specific buffer from the parameter)